internal/primer/primer.go

```diff
@@ -18,15 +18,17 @@ type Values struct {
 	prompt      prompt.Prompter
 	subshell    subshell.SubShell
 	conditional *constraints.Conditional
+	args        []string
 }
 
-func New(project *project.Project, output output.Outputer, auth *authentication.Auth, prompt prompt.Prompter, subshell subshell.SubShell, conditional *constraints.Conditional) *Values {
+func New(project *project.Project, output output.Outputer, auth *authentication.Auth, prompt prompt.Prompter, subshell subshell.SubShell, conditional *constraints.Conditional, args []string) *Values {
```

Comment:
Primer is for passing persistent variables around. It's not a catchall for passing arguments. You shouldn't need to pass args this way at all, captain/cobra already has access to this. It's already part of the architecture.

Reply:
The problem with accessing the args from captain/cobra is that by that time the flag are already parsed. I've removed the args from primer and moved the delimiter check.